### PR TITLE
Updated reliable delivery tutorial

### PR DESCRIPTION
**Pull request by MalshaL**
## Purpose
Updated reliable delivery tutorial.

### CODE REVIEW

File: docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md [hunk 1]
```diff
@@ -1,98 +1,100 @@
...
 
-Upon receiving the response from the backend, user is allowed to perform any action upon the response as per the design
-of *messageStore* module. This action is passed as a Lambda (function pointer) to the message processor when it is created.
-For simplicity, in this guide it will just log the response from the backend to the Ballerina log file.
+<!-- INCLUDE_MD: ../../../../tutorial-prerequisites.md -->
+* A message broker to store messages in a queue. You can download and install Apache ActiveMQ from 
+[here](http://activemq.apache.org/getting-started.html).
 
-The following diagram illustrates the scenario in high level:
+## Get the code
```
Comment (samgnaniah) [inline docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md]: There is a way to insert this section automatically. This is similar to how you include prerequisites. Please check.

File: docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md [hunk 2]
```diff
@@ -1,98 +1,100 @@
...
-If the message is successfully stored, HTTP caller will get *202-accepted* message, or else *500-internal server error*. In this case,
-the caller is notified of successful message acceptance.
+####3. Implement message processor
 
-### Developing message forwarding service
+Message processor will poll the message broker and listen for any new messages on the queue. If new messages are added, 
+it will dequeue the message, and forward it to the backend endpoint for further processing.
 
-When configuring **MessageForwardingProcessor**, there is a few things you need to consider.
+When configuring the `MessageForwardingProcessor`, there is a few things you need to consider.
```
Comment (samgnaniah) [inline docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md]:
```suggestion
When configuring the `MessageForwardingProcessor`, there are a few things you need to consider.
```

File: docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md [hunk 3]
```diff
@@ -113,144 +115,105 @@ When configuring **MessageForwardingProcessor**, there is a few things you need
...
 connection establishment failed for the broker).
 
-**message_forwarder.bal**
+<!-- INCLUDE_CODE: src/message_processor/msg_processor.bal -->
 
-<!-- INCLUDE_CODE: guide/message_forwarder.bal-->
+####4. Implement the backend service
 
-## Deployment
+The backend service exposes `/sales/save-order` endpoint as a `POST` resource, which accepts a Json message. It returns
```
Comment (samgnaniah) [inline docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md]:
```suggestion
The backend service exposes `/sales/save-order` endpoint as a `POST` resource, which accepts a JSON message. It returns 
```

File: docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md [hunk 4]
```diff
@@ -113,144 +115,105 @@ When configuring **MessageForwardingProcessor**, there is a few things you need
...
-Note that upon forwarding failure, the message forwarding processor will behave in the following priority order:
+1. **in-order message delivery** : If only one message processor is picking messages from the particular store, in-order 
+message delivery pattern is also achieved. Messages are delivered to the endpoint in a reliable manner keeping the order 
+of the messages they were put to the store.
 
-1. If {`deactivateOnFail: true`} is configured it will be deactivated and message polling will stop. Message will remain on the store.
-2. If {`DLCStore: dlcStoreClient`} is configured message will be routed to that queue pointed by that client and will be removed from
-   the original store.
-3. If non of the above is specified, message processor will drop the message and pick the next message in the store to try.
+2. **throttling** : Consider a legacy backend which can process messages only up to 100 TPS. This service is exposed to
```
Comment (samgnaniah) [inline docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md]:
```suggestion
2. **throttling** : Consider a legacy backend that can process messages only up to 100 TPS. This service is exposed to 
```

File: docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md (hunk 3 above)
Comment (MalshaL) [reply docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md]: Fixed

File: docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md (hunk 2 above)
Comment (MalshaL) [reply docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md]: Fixed

File: docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md (hunk 4 above)
Comment (MalshaL) [reply docs/content/src/learn/tutorials/messaging-integrations/reliable-delivery/README.md]: Fixed

**Pull request merged**